File: app/src/main/java/com/examples/memorizingpager_demo/MessagePagerAdapter.java
package com.examples.memorizingpager_demo;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

class MessagePagerAdapter extends FragmentPagerAdapter {

  private static final int NUM_PAGES = 4;

  MessagePagerAdapter(FragmentManager fm) {
    super(fm);
  }

  @Override
  public Fragment getItem(int position) {
    return MessageFragment.newInstance(position);
  }

  @Override
  public int getCount() {
    return NUM_PAGES;
  }

}

File: app/src/main/java/com/examples/memorizingpager_demo/MessagePagerAdapter.kt
package com.examples.memorizingpager_demo

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter

internal class MessagePagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm) {

    override fun getItem(position: Int): Fragment = MessageFragment.newInstance(position)
    override fun getCount(): Int = NUM_PAGES

    companion object {

        private val NUM_PAGES = 4
    }

}